### PR TITLE
Add end-to-end test for device registration

### PR DESCRIPTION
**Pull request by jonathanlloyd**

### CODE REVIEW

File: end-to-end-tests/sdk-start.test.js [hunk 1]
```diff
@@ -11,28 +11,73 @@ beforeAll(() => {
...
   await chromeDriver.wait(() => {
-    return chromeDriver.getTitle()
-      .then(title => title.includes('Test Page'))
+    return chromeDriver.getTitle().then(title => title.includes('Test Page'));
   }, 2000);
 
-  const result = await chromeDriver.executeScript(() => {
-    return document.title;
+  const initialDeviceId = await chromeDriver.executeAsyncScript(() => {
+    var callback = arguments[arguments.length - 1];
```
Comment (minaorangina) [inline end-to-end-tests/sdk-start.test.js]: prefer `let` or `const`

File: end-to-end-tests/sdk-start.test.js [hunk 2]
```diff
@@ -11,28 +11,73 @@ beforeAll(() => {
...
 });
 
+test('SDK should remember the device ID', async () => {
+  await chromeDriver.get('http://localhost:3000');
+  await chromeDriver.wait(() => {
+    return chromeDriver.getTitle().then(title => title.includes('Test Page'));
+  }, 2000);
+
+  const initialDeviceId = await chromeDriver.executeAsyncScript(() => {
+    var callback = arguments[arguments.length - 1];
```
Comment (minaorangina) [inline end-to-end-tests/sdk-start.test.js]: prefer `let` or `const`

File: end-to-end-tests/sdk-start.test.js [hunk 3]
```diff
@@ -11,28 +11,73 @@ beforeAll(() => {
...
+      .catch(e => callback(e.message));
+  });
+
+  await chromeDriver.get('http://localhost:3000');
+  await chromeDriver.wait(() => {
+    return chromeDriver.getTitle().then(title => title.includes('Test Page'));
+  }, 2000);
+
+  const reloadedDeviceId = await chromeDriver.executeAsyncScript(() => {
+    var callback = arguments[arguments.length - 1];
```
Comment (minaorangina) [inline end-to-end-tests/sdk-start.test.js]: prefer `let` or `const`

**Comment by minaorangina** [comment]
Just sort out the `var`s and it's fine

File: end-to-end-tests/sdk-start.test.js (hunk 1 above)
Comment (lukabratos) [reply end-to-end-tests/sdk-start.test.js]: Fixed in 9a26dc8.

File: end-to-end-tests/sdk-start.test.js (hunk 2 above)
Comment (lukabratos) [reply end-to-end-tests/sdk-start.test.js]: Fixed in 9a26dc8.

File: end-to-end-tests/sdk-start.test.js (hunk 3 above)
Comment (lukabratos) [reply end-to-end-tests/sdk-start.test.js]: Fixed in 9a26dc8.

**Pull request merged**